packages/types/src/parser-options.ts

```diff
@@ -51,8 +51,8 @@ interface ParserOptions {
   extraFileExtensions?: string[];
   filePath?: string;
   loc?: boolean;
-  program?: Program;
-  project?: string | string[] | true;
+  program?: Program | null;
+  project?: string | string[] | true | null;
```

Comment:
Is this needed for this PR?

It's also weird to me IMO to have `null` be an option only for some of these.

Reply:
I added null because it's an explicit value to set these to off that won't get weirdly merged by ESLint's cascading config. It also means the configs can be json-serialized without losing any data (`undefined` tells a json stringifier to omit the property).

It just means we can be sure these configs are robust.

I also considered using `[]`, but IIRC ESLint's cascading config logic will merge arrays together. Which means that `[]` is the same as not specifying a value at all.